pkg/dao/interfaces.go

```diff
@@ -70,9 +73,11 @@ type RepositoryConfigDao interface {
 //go:generate mockery --name RpmDao --filename rpms_mock.go --inpackage
 type RpmDao interface {
 	List(orgID string, uuidRepo string, limit int, offset int, search string, sortBy string) (api.RepositoryRpmCollectionResponse, int64, error)
+	SearchSnapshotRpms(orgId string, request api.SnapshotSearchRpmRequest) ([]api.SearchRpmResponse, error)
 	Search(orgID string, request api.SearchRpmRequest) ([]api.SearchRpmResponse, error)
 	InsertForRepository(repoUuid string, pkgs []yum.Package) (int64, error)
 	OrphanCleanup() error
+	WithContext(ctx context.Context) RpmDao
```

Comment:
I know there's another place we set context using this method. But I don't remember, was there a benefit to setting context this way vs passing it as the first argument in each method? I'm just curious because the latter is more go-idiomatic and I don't remember why we chose to do it this way.

Reply:
i'm fine doing it the other way.